src/benchmarks/gc/GC.Infrastructure/GC.Infrastructure/Commands/GCPerfSim/GCPerfSimFunctionalCommand.cs

```diff
@@ -0,0 +1,336 @@
+            {
+                GCPerfSimConfiguration gcperfsimConfiguration  = 
+                    GCPerfSimConfigurationParser.Parse(gcperfsimConfigurationFileName);
+
+                Stopwatch sw = new();
+                sw.Start();
+
+                AnsiConsole.Write(new Rule($"{gcperfsimConfiguration.Name}"));
+                AnsiConsole.WriteLine();
+
+                // run the test
+                GCPerfSimResults gcperfsimResult = GCPerfSimCommand.RunGCPerfSim(gcperfsimConfiguration, settings.Server);
+                string yamlFileName = Path.GetFileName(gcperfsimConfigurationFileName);
+                yamlFileResultMap[yamlFileName] = gcperfsimResult;
+
+                
+                sw.Stop();
+                AnsiConsole.WriteLine($"Time to execute Msec: {sw.ElapsedMilliseconds}");
+            }
+            // Look at the RunCommand from 99 onwards to see how these are generated.
+
+            // IV. Based on the results of the functional tests, we'd want to generate a report.
+            // Looking at all the runs data, we'd want to aggregate and create a markdown table of the following type:
+            // Output should live in Results.md in the output folder.
+            // TODO: If the run fails, what are the commands, failures and any debugging information that'll be helpful.
+            // | Yaml | Scenario | Pass / Fail  |
+            // | ---- | -------- | ------------ |
+            // | Normal_Server.yaml | 0gb | Pass |
+            // | Normal_Workstation.yaml | 2gb_pinning | Fail |
+            string markdownPath = Path.Combine(configuration.output_path, "Results.md");
+            using (StreamWriter resultWriter = new StreamWriter(markdownPath))
+            {
+                resultWriter.WriteLine("| Yaml | Scenario | Pass / Fail  |");
+                resultWriter.WriteLine("| ---- | -------- | ------------ |");
+
+
+                StringBuilder resultContent = new StringBuilder();
+
+                foreach (var yamlFileNameResultPair in yamlFileResultMap)
+                {
+                    string yamlFileName = yamlFileNameResultPair.Key;
+                    GCPerfSimResults gcperfsimResult = yamlFileNameResultPair.Value;
+                    foreach (var executionDetail in gcperfsimResult.ExecutionDetails)
+                    {
+                        string scenario = executionDetail.Key;
+                        string testResult = executionDetail.Value.HasFailed == false ? "pass" : "fail";
+                        resultContent.AppendLine($"| {yamlFileName} | {scenario} | {testResult} |");
+                    }
+                }
+
+                resultWriter.Write(resultContent.ToString());
+
+                foreach (var yamlFileNameResultPair in yamlFileResultMap)
+                {
+                    resultWriter.AddIncompleteTestsSection(new (yamlFileNameResultPair.Value.ExecutionDetails));
```

Comment:
Modify the method so that we print out the Name of the Yaml file e.g., 

```
# Incomplete Tests : LowMemoryContainer.yaml
```

Secondly, don't include this section if we don't have any failed tests.